### PR TITLE
implement `create` query lookup strategy

### PR DESCRIPTION
**Pull request by kovrus**

### CODE REVIEW

File: src/test/java/org/springframework/data/crate/query/SimpleQueryCrateHandlerTest.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
 import org.junit.BeforeClass;
 import org.junit.Test;
 import org.springframework.data.crate.core.CrateActionResponseHandler;
+import org.springframework.data.crate.query.execution.SimpleQueryCrateHandler;
```
Comment (memcmp) [inline src/test/java/org/springframework/data/crate/query/SimpleQueryCrateHandlerTest.java]: remove or was it broken before?

Comment (kovrus) [inline src/test/java/org/springframework/data/crate/query/SimpleQueryCrateHandlerTest.java]: i moved it into the other package

**Comment by memcmp** [comment]
LGTM

File: src/test/java/org/springframework/data/crate/repository/support/CreateLookupStrategyIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    @Test
+    public void testSimpleNotCollectionQuery() {
+        Person person = repository.readByEmail("user5@test.com");
+        assertThat(person.getAge(), is(50));
+    }
+
+    @Test
+    public void testCollectionQueryWithAnd() {
+        Person persons = repository.findByNameAndEmail("user3", "user3@test.com");
```
Comment (chaudum) [inline src/test/java/org/springframework/data/crate/repository/support/CreateLookupStrategyIntegrationTest.java]: person

**Comment by chaudum** [comment]
no unit tests for `PartTreeCrateRepositoryQuery`?

**Comment by chaudum** [comment]
also check travis!

File: src/main/java/org/springframework/data/crate/query/PartTreeCrateRepositoryQuery.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    private final Class<?> domainClass;
+    private final PartTree tree;
+    private final Table table;
+    private final MappingContext<?, CratePersistentProperty> mappingContext;
+
+    public PartTreeCrateRepositoryQuery(CrateQueryMethod queryMethod, CrateOperations crateOperations) {
+        super(queryMethod, crateOperations);
+        this.domainClass = queryMethod.getEntityInformation().getJavaType();
+        this.tree = new PartTree(queryMethod.getName(), domainClass);
+        this.table = domainClass.getAnnotation(Table.class);
```
Comment (chaudum) [inline src/main/java/org/springframework/data/crate/query/PartTreeCrateRepositoryQuery.java]: `table` is only used to get it's name
does not require to make a field, rather make a field `<Set>String roots`
this reduces the overhead of generating the set for every query

File: src/main/java/org/springframework/data/crate/query/PartTreeCrateRepositoryQuery.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import org.springframework.data.repository.query.ParametersParameterAccessor;
+import org.springframework.data.repository.query.parser.PartTree;
+
+import java.util.Set;
+
+import static org.springframework.util.StringUtils.hasText;
+
+public class PartTreeCrateRepositoryQuery extends CrateRepositoryQuery {
+
+    private final Class<?> domainClass;
```
Comment (chaudum) [inline src/main/java/org/springframework/data/crate/query/PartTreeCrateRepositoryQuery.java]: not required as a field

**Comment by chaudum** [comment]
ack

**Pull request merged**